File: app/src/main/java/cat/teknos/berry/model/PlaylistManager.java
package cat.teknos.berry.model;

import android.content.Context;
import android.media.MediaPlayer;

public class PlaylistManager implements MediaPlayer.OnCompletionListener {
    private MediaPlayer[] playlist;
    private int currentSongIndex = 0;

    public PlaylistManager(Context context, int[] songResources) {
        initPlaylist(context, songResources);
    }

    private void initPlaylist(Context context, int[] songResources) {
        playlist = new MediaPlayer[songResources.length];

        for (int i = 0; i < songResources.length; i++) {
            playlist[i] = MediaPlayer.create(context, songResources[i]);
            playlist[i].setOnCompletionListener(this);
        }
    }

    public void start() {
        playCurrentSong();
    }

    public void release() {
        for (MediaPlayer mediaPlayer : playlist) {
            if (mediaPlayer != null) {
                mediaPlayer.release();
            }
        }
    }

    private void playCurrentSong() {
        if (currentSongIndex >= 0 && currentSongIndex < playlist.length) {
            MediaPlayer currentSong = playlist[currentSongIndex];
            if (currentSong != null) {
                currentSong.start();
            }
        }
    }

    private void playNextSong() {
        currentSongIndex = (currentSongIndex + 1) % playlist.length;
        playCurrentSong();
    }

    @Override
    public void onCompletion(MediaPlayer mp) {
        playNextSong();
    }
}

File: app/src/main/java/cat/teknos/berry/model/PlaylistManager.kt
package cat.teknos.berry.model

import android.content.Context
import android.media.MediaPlayer
import android.media.MediaPlayer.OnCompletionListener

class PlaylistManager(context: Context, songResources: IntArray) : OnCompletionListener {
    private lateinit var playlist: Array<MediaPlayer?>
    private var currentSongIndex = 0

    init {
        initPlaylist(context, songResources)
    }

    private fun initPlaylist(context: Context, songResources: IntArray) {
        playlist = arrayOfNulls(songResources.size)

        for (i in songResources.indices) {
            playlist[i] = MediaPlayer.create(context, songResources[i])
            val volume = 0.5f
            playlist[i]?.setVolume(volume, volume)
            playlist[i]?.setOnCompletionListener(this)
        }
    }

    fun start() {
        playCurrentSong()
    }

    fun pause() {
        pauseCurrentSong()
    }

    private fun pauseCurrentSong() {
        if (currentSongIndex >= 0 && currentSongIndex < playlist.size) {
            val currentSong = playlist[currentSongIndex]
            if (currentSong != null && currentSong.isPlaying) {
                currentSong.pause()
            }
        }
    }

    fun release() {
        for (mediaPlayer in playlist) {
            mediaPlayer?.release()
        }
    }

    private fun playCurrentSong() {
        if (currentSongIndex >= 0 && currentSongIndex < playlist.size) {
            val currentSong = playlist[currentSongIndex]
            currentSong?.start()
        }
    }

    private fun playNextSong() {
        currentSongIndex = (currentSongIndex + 1) % playlist.size
        playCurrentSong()
    }

    override fun onCompletion(mp: MediaPlayer) {
        playNextSong()
    }
}